.github/actions/start-opensearch-with-one-plugin/action.yml

```diff
@@ -0,0 +1,126 @@
+
+inputs:
+  opensearch-version:
+    description: 'The version of OpenSearch that should be used, e.g "3.0.0"'
+    required: true
+
+  plugin-name:
+    description: 'The the name of the plugin to use, such as security-dashboards-plugin'
+    required: true
+
+  plugin-start-script:
+    description: 'The file name for the configuration script for the plugin such as install_demo_configurations -- may not be needed for every plugin'
+    required: false
+
+  plugin-zip:
+    description: 'The name of the zip file for the plugin hosted on docker-host i.e. security-plugin.zip '
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+
+    # Configure longpath names if on Windows
+    - name: Enable Longpaths if on Windows
+      if: ${{ runner.os == 'Windows' }}
+      run: git config --system core.longpaths true
+      shell: pwsh
+
+    # Download OpenSearch
+    - uses: peternied/download-file@v1
+      if: ${{ runner.os == 'Windows' }}
+      with:
+        url: https://artifacts.opensearch.org/snapshots/core/opensearch/${{ inputs.opensearch-version }}-SNAPSHOT/opensearch-min-${{ inputs.opensearch-version }}-SNAPSHOT-windows-x64-latest.zip
+
+    - uses: peternied/download-file@v1
+      if: ${{ runner.os == 'Linux' }}
+      with:
+        url: https://ci.opensearch.org/ci/dbc/distribution-build-opensearch/${{ inputs.opensearch-version }}/latest/linux/x64/tar/builds/opensearch/dist/opensearch-min-${{ inputs.opensearch-version }}-linux-x64.tar.gz
+
+    # Extract downloaded zip
+    - name: Extract downloaded zip for Linux
+      if: ${{ runner.os == 'Linux' }}
+      run: |
+        tar -xzf opensearch-*.tar.gz
+        rm -f opensearch-*.tar.gz
+      shell: bash
+
+    - name: Extract downloaded zip for Windows
+      if: ${{ runner.os == 'Windows' }}
+      run: |
+        tar -xzf opensearch-min-${{ inputs.opensearch-version }}-SNAPSHOT-windows-x64-latest.zip
+        del opensearch-min-${{ inputs.opensearch-version }}-SNAPSHOT-windows-x64-latest.zip
+      shell: pwsh
+
+    # Move and rename the plugin for installation
+    - name: Move and rename the plugin for installation
+      run: mv ./build/distributions/${{ inputs.plugin-name }}-*.zip ${{ inputs.plugin-name }}.zip
+      shell: bash
+
+    # Install the plugin, runs its start-script, and start the OpenSearch server
+    - name: Install Plugin into OpenSearch for Linux
+      if: ${{ runner.os == 'Linux'}}
+      run: |
+        cat > os-ep.sh <<EOF
+        yes | opensearch-plugin install file:///docker-host/${{ inputs.plugin-zip }}.zip
+        chmod +x plugins/${{ inputs.plugin-name }}/tools/${{ inputs.plugin-start-script }}.sh
+        yes | plugins/${{ inputs.plugin-name }}/tools/${{ inputs.plugin-start-script }}.sh
+        chown 1001:1001 -R /opensearch
+        su -c "/opensearch/bin/opensearch" -s /bin/bash opensearch
+        EOF
+        docker build -t opensearch-test:latest -f- . <<EOF
```

Comment:
Curious, is tagging this image with `opensearch-test:latest` needed?

Reply:
No, if you do not specify the a version tag for the image it will pull the latest code. That being said, I do think for readability it is nice to have the image tag there since people are most likely to be used to seeing docker images tagged like this. 

I can remove it if you prefer however.

Reply:
This command is building the container with an inline build script. Tagging the image is useful when publishing to docker hub, but doesn't look like its being used here. I know it wasn't introduced with this PR, but think its something worth pointing out.